From 73622adeab66b50d614c50429d17a9eaf2c2476a Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Fri, 5 Jan 2018 17:41:03 -0800
Subject: [PATCH 08/10] dma: xilinx: dpdma: Fix the macro argument reuse
 warning

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Fix a warning:
CHECK: Macro argument reuse 'x' - possible side-effects?
+#define IN_RANGE(x, min, max) ((x) >= (min) && (x) <= (max))

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 7d6802089babac45b18d92c73dbaa13080c794f6)
---
 drivers/dma/xilinx/xilinx_dpdma.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/dma/xilinx/xilinx_dpdma.c b/drivers/dma/xilinx/xilinx_dpdma.c
index f9b0e94..146c0d2 100644
--- a/drivers/dma/xilinx/xilinx_dpdma.c
+++ b/drivers/dma/xilinx/xilinx_dpdma.c
@@ -340,7 +340,9 @@ struct xilinx_dpdma_device {
 #ifdef CONFIG_XILINX_DPDMA_DEBUG_FS
 #define XILINX_DPDMA_DEBUGFS_READ_MAX_SIZE	32
 #define XILINX_DPDMA_DEBUGFS_UINT16_MAX_STR	"65535"
-#define IN_RANGE(x, min, max) ((x) >= (min) && (x) <= (max))
+#define IN_RANGE(x, min, max) ({		\
+		typeof(x) _x = (x);		\
+		_x >= (min) && _x <= (max); })
 
 /* Match xilinx_dpdma_testcases vs dpdma_debugfs_reqs[] entry */
 enum xilinx_dpdma_testcases {
-- 
1.8.5.2.233.g932f7e4

